pkg/reconciler/instances/istio/reset/data/gatherer.go

```diff
@@ -56,12 +58,30 @@ func (i *DefaultGatherer) GetPodsWithDifferentImage(inputPodsList v1.PodList, im
 	outputPodsList.Items = []v1.Pod{}
 
 	for _, pod := range inputPodsList.Items {
+		if _, ok := pod.Annotations["sidecar.istio.io/status"]; !ok || !isPodReady(pod) {
+			continue
+		}
+
+		type IstioStatusStruct struct {
+			Containers []string `json:"containers"`
+		}
+		istioStatus := IstioStatusStruct{}
+		err := json.Unmarshal([]byte(fmt.Sprintf("{%s}", pod.Annotations["sidecar.istio.io/status"])), &istioStatus)
+		if err != nil {
+			continue
+		}
```

Comment:
You could wrap it in some func and give it a meaningful name such as: `retrieveIstioContainerNames`